apps/rules/routing_rules.py

```diff
@@ -210,7 +210,12 @@ def apply_routing_scheme(self, ingest_item, provider, routing_scheme):
             if filters_service.does_match(content_filter, ingest_item):
                 if rule.get('actions', {}).get('preserve_desk', False) and ingest_item.get('task', {}).get('desk'):
                     desk = get_resource_service('desks').find_one(req=None, _id=ingest_item['task']['desk'])
-                    self.__fetch(ingest_item, [{'desk': desk[config.ID_FIELD], 'stage': desk['incoming_stage']}])
+                    if ingest_item.get('task', {}).get('stage'):
+                        stage = get_resource_service('stages').find_one(req=None, _id=ingest_item['task']['stage'])
```

Comment:
Is this line needed? We already know the stage id from ingest_item['task']['stage']...